grails-app/migrations/views/order-state-view.sql

```diff
@@ -0,0 +1,160 @@
+CREATE OR REPLACE VIEW order_item_status AS
+    SELECT order_id,
+        order_number,
+        product_code,
+        quantity_ordered,
+        quantity_shipped,
+        order_status,
+        (quantity_ordered > 0)                 AS shipment_ordered,
+        (quantity_shipped >= quantity_ordered) AS shipped
+    FROM (
+        SELECT order.id                 AS order_id,
+            `order`.order_number        AS order_number,
+            order.status                AS order_status,
+            order_item.id               AS order_item_id,
+            product.product_code        AS product_code,
+            SUM(order_item.quantity)    AS quantity_ordered,
+            SUM(shipment_item.quantity) AS quantity_shipped
+        FROM `order`
+            LEFT OUTER JOIN order_item ON order.id = order_item.order_id
+            LEFT OUTER JOIN product ON order_item.product_id = product.id
+            LEFT OUTER JOIN order_shipment ON order_item.id = order_shipment.order_item_id
+            LEFT OUTER JOIN shipment_item ON shipment_item.id = order_shipment.shipment_item_id
+        WHERE `order`.order_type_code = 'PURCHASE_ORDER' AND `order`.status NOT IN ('COMPLETED', 'CANCELLED') AND order_item.order_item_status_code != 'CANCELLED'
```

Comment:
Can you separate these three conditions onto separate lines so they are more readable? And I think we'd like to keep orders which are in those two states, no?

Reply:
In other words, I would like one row for every PO in the database. Therefore, Canceled and Completed are valid states.

Reply:
Done